src/nspawn/nspawn.c

```diff
@@ -3933,7 +3933,7 @@ int main(int argc, char *argv[]) {
                                 log_error_errno(r, "Directory tree %s is currently busy.", arg_directory);
                                 goto finish;
                         }
-                        if (r < 0) {
+                        if (r < 0 && !(arg_read_only && r == -EROFS)) {
```

Comment:
hmm, note that image_path_lock() creates two locks, one next to the image and one in /run. I think we should always create the latter, even if the former fails.

Hence, please change the implementation of image_path_lock() to handle EROFS on the image lock gracefully (and return a NULL lock in that case). Also, please add a debug logging message in that case so that this remains debuggable.